hudi-spark/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -145,7 +146,7 @@ private[hudi] object HoodieSparkSqlWriter {
         log.info("new batch has no new records, skipping...")
         return (true, common.util.Option.empty())
       }
-      commitTime = client.startCommit()
+      client.startCommitWithTime(commitTime)
```

Comment:
@vinothchandar : does "rollbackPendingCommits" matter here? only diff I see between client.startCommit() and callingclient.startcommitWithTime(time) is rolling back pending commits. if that is not required in this path, this patch is good.

Reply:
Thanks for reviewing! 
In the latest master, both startCommit() and startCommitWithTime(time) are rolling back the pending commits. So I think those two functions have the same behavior except the `String commitTime`.
https://github.com/apache/incubator-hudi/blob/master/hudi-client/src/main/java/org/apache/hudi/HoodieWriteClient.java#L1009

Reply:
cool. can you please rebase with upstream/master and update the commit.

Reply:
That’s strange... This commit was created from the latest master and I don’t see any diff from master. Could that be any conflict?